index.bs

```diff
@@ -96,6 +106,38 @@ run the following steps:
 
 </div>
 
+## Permissions ## {#permissions}
+
+Note: By default entries inherit their permissions from their parent. If an entry doesn't
+have a parent, permissions are by default granted, and requesting permissions is a no-op.
+Entries returned by {{chooseFileSystemEntries()}} override these steps, providing stricter
+access control for those entries.
+
+Issue: Figure out some way to integrate this better with the permissions API [[permissions]],
+for example by making the entry be a part of the PermissionDescriptor.
+
+<div algorithm>
+An [=/entry=] |entry| has associated <dfn for=entry>query permission steps</dfn>,
+which take a {{FileSystemHandlePermissionDescriptor}} and return a {{PermissionState}}.
+Unless specified otherwise, these steps are:
+1. Let |parent| be |entry|'s [=entry/parent=].
+1. If |parent| is null, return {{PermissionState/"granted"}}.
+1. Return the result of running |parent|'s [=query permission steps=].
+
+Note: These steps return the current state synchronously, however these steps are only
+invoked from "in parallel" sections of algorithms, so this doesn't have to be implemented
+in a synchronous manner.
+</div>
+
+<div algorithm>
+An [=/entry=] |entry| also has associated <dfn for=entry>request permission steps</dfn>,
```

Comment:
Maybe note that there's no return value, but the permission state may be updated and can be queried.